tests/class_test.py

```diff
@@ -279,6 +279,29 @@ def test_cannot_remove_non_existing_member():
         p1.remove('y')
 
 
+def test_set_equal_basic_types_returns_original_instance():
+    p1 = Point(x=1)
+
+    assert p1.set('x', 1) is p1
+    assert p1.set(x=1) is p1
+
+
+def test_set_equal_pyrsistent_types_returns_original_instance():
+    l1 = Line(p1=Point(x=2, y=1), p2=Point(x=20, y=10))
+
+    assert l1.set(p1=Point(x=2, y=1)) is l1
+
+
+def test_set_mutable_types_only_returns_original_if_exact_object():
+    class MyClass(PClass):
+        a = field()
+
+    d1 = dict(x=1)
+    c1 = MyClass(a=d1)
+    assert c1.set(a=d1) is c1
+    assert c1.set(a=dict(x=1)) is not c1
```

Comment:
And if you like the alternative implementation (above) perhaps this test should be a general test for set when the new and old values are the same object.